tests/yadism/test_tmc.py

```diff
@@ -159,10 +159,10 @@ def isdelta(pdf):  # assert F2 = pdf
             isdelta(pdf_lin)
             # int_const = int_xi^1 du = 1-xi
             integral_with_pdf = np.matmul(res_const.orders[lo][0][0], pdf_lin)
-            assert pytest.approx(integral_with_pdf, 1 / 1000.0) == c * (1.0 - obj.xi)
+            np.testing.assert_allclose(integral_with_pdf, c * (1.0 - obj.xi), rtol=3e-2)
```

Comment:
I have no clue why I had to change this in order to pass...

Reply:
If you want, you can add `rtol` also using `pytest.approx`. We are using `np.testing`  mainly when arrays are involved, but [`pytest.approx`](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-approx) actually works also in that case (so it is a complete replacement)

However, yes, it is weird...

Reply:
Might it be that we change from linear interpolation to log interpolation? because now https://github.com/NNPDF/eko/blob/e3466fae35fd245c7e1a99158d8582f590276555/src/eko/interpolation.py#L501-L505 by default we're  log ... before there was the explict `log=False` argument (see https://github.com/NNPDF/eko/blob/01efee1d27d215e895dd055debff8905af3d548c/src/eko/interpolation.py#L439) in line 50

Note that linear here is fine, since we're faking in any case with 3 points in the linear regime. So to recover make `xg` linear, I guess ..

Reply:
You're right. Then, I'd try to revert and check if the problem persists.
Can you do this? @andreab1997 @giacomomagni

Reply:
Thanks @felixhekhorn for the suggestion, indeed this solved the problem!!  :)